fix: LEAP-509: Optimize regular expression for improved performance

juliosgarbi opened the pull request:
This PR addresses inefficient regular expressions in the code, as identified by GitHub's code scan. In particular, the regular expression "(?""|[^""]+)*" has been causing performance issues and potential security risks. The goal is to enhance performance and mitigate the risk of Denial of Service ("DoS") attacks. https://github.com/HumanSignal/label-studio/security/code-scanning/760

### PR fulfills these requirements
- [x] Commit message(s) and PR title follows the format `[fix|feat|ci|chore|doc]: TICKET-ID: Short description of change made` ex. `fix: DEV-XXXX: Removed inconsistent code usage causing intermittent errors`
- [ ] Tests for the changes have been added/updated (for bug fixes/features)
- [ ] Docs have been added/updated (for bug fixes/features)
- [x] Best efforts were made to ensure docs/code are concise and coherent (checked for spelling/grammatical errors, commented out code, debug logs etc.)
- [x] Self-reviewed and ran all changes on a local instance (for bug fixes/features)



#### Change has impacts in these area(s)
- [ ] Product design
- [ ] Backend (Database)
- [ ] Backend (API)
- [x] Frontend



#### What does this fix?
By optimizing the regular expression, this change improves code performance and reduces the risk of DoS attacks. It ensures that the regular expression matching process is more efficient and not negatively affecting the application's responsiveness.


#### Does this change affect performance?
Yes, this change positively affects performance. The replacement of the inefficient regular expression with the optimized regex "(?:""|[^"])*" significantly improves matching efficiency. It reduces the risk of performance issues, making the regular expression matching process more efficient and responsive.
- https://github.com/HumanSignal/label-studio-frontend/pull/1673

juliosgarbi committed:
fix: LEAP-509: Optimize regular expression for improved performance

robot-ci-heartex committed:
ci: Build frontend

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/7585393196

juliosgarbi committed:
fix regex to match the scope

juliosgarbi committed:
Merge branch 'fb-leap-509' of https://github.com/HumanSignal/label-studio into fb-leap-509

robot-ci-heartex committed:
ci: Build frontend

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/7586535924

hlomzik commented on web/libs/editor/src/utils/data.js:
```diff
@@ -72,7 +72,7 @@ export const parseCSV = (text, separator = 'auto') => {
-      '"(""|[^"]+)*"', // quoted text with possible quoted quotes inside it ("not a ""value""")
+      '"(?:""|[^"])*"', // quoted text with possible quoted quotes inside it ("not a ""value""")
       `[^"${separator}]+`, // usual value, no quotes, between separators
       `(?=${separator}(${separator}|$))`, // empty value in the middle or at the end of string
```
```suggestion
      `(?=${separator}(?:${separator}|$))`, // empty value in the middle or at the end of string
```

juliosgarbi committed:
Update web/libs/editor/src/utils/data.js

Co-authored-by: hlomzik <hlomzik@gmail.com>

robot-ci-heartex committed:
ci: Build frontend

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/7611143402

juliosgarbi committed:
Merge branch 'develop' into 'fb-leap-509'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/7611551866

juliosgarbi committed:
[submodules] Copy src HumanSignal/label-studio-frontend

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/7611551866

juliosgarbi committed:
Merge branch 'develop' into 'fb-leap-509'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/7643711911

juliosgarbi committed:
[submodules] Copy src HumanSignal/label-studio-frontend

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/7643711911

robot-ci-heartex committed:
ci: Build frontend

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/7643735748

juliosgarbi committed:
Merge branch 'develop' into 'fb-leap-509'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/7646041615

juliosgarbi committed:
[submodules] Copy src HumanSignal/label-studio-frontend

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/7646041615
